src-electron/generator/matter/controller/python/templates/helper.js

```diff
@@ -43,6 +49,58 @@ function asPythonType(zclType) {
   }
 }
 
+/**
+ *
+ * Available options:
+ * - All options passed to this helper are considered as overrides for
+ * zcl types
+ * for eg: (as_underlying_python_zcl_type type clusterId boolean='bool')
+ * will return "bool" for "boolean" type
+ * @param {*} type
+ * @param {*} clusterId
+ * @param {*} options
+ * @returns The corresponding python data type for a zcl data type.
+ */
+async function as_underlying_python_zcl_type(type, clusterId, options) {
+  let hash = options.hash;
+  // Overwrite any type with the one coming from the template options
+  // Eg: {{as_underlying_python_zcl_type type [clusterId] boolean='bool'}}
+  // Here all types named 'boolean' will be returned as 'bool'
+  if (type in hash) {
+    return hash[type];
+  }
+
+  // Get ZCL Data Type from the db
+  const packageIds = await templateUtil.ensureZclPackageIds(this);
+  let dataType = await queryZcl.selectDataTypeByNameAndClusterId(
+    this.global.db,
+    type,
+    clusterId,
+    packageIds
+  );
+  if (
+    dataType.discriminatorName.toLowerCase() == dbEnum.zclType.bitmap ||
+    dataType.discriminatorName.toLowerCase() == dbEnum.zclType.enum ||
+    dataType.discriminatorName.toLowerCase() == dbEnum.zclType.number
+  ) {
+    // returning nothing for floats
```

Comment:
Why?  The comment should explain the "why", not the "what"....

Reply:
I don't know, the deprecated helper `asPythonCType` does not seem to say why

Reply:
So ... I don't know why `asPythonCType` does what it does, but it's not used anywhere, right?  

I'm assuming you meant `asPythonType`.  If so: I suspect that thing is just broken.  What we should probably have here is a comment that (1) says this is being doen for compat with `asPythonType` and (2) pointing to the issue tracking this brokenness...

Reply:
done